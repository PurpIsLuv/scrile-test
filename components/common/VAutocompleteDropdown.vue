<template>
  <div class="v-autocomplete-dropdown">
    <ol>
      <v-autocomplete-dropdown-item
        v-for="user in users"
        :key="user.id"
        :name="user.name"
        :username="user.username"
        :src="user.image.url"
        :alt="user.image.title"
        @click="$emit('selected', user.id)"
      />
    </ol>
  </div>
</template>

<script>
export default {
  props: {
    users: {
      type: Array,
      default: () => []
    }
  }
}
</script>

<style lang="scss" scoped>
.v-autocomplete-dropdown {
  max-height: 343px;
  overflow-y: scroll;
  scrollbar-width: none;
  &::-webkit-scrollbar {
    display: none;
  }
  ul {
    list-style-type: none;
  }
}
</style>
