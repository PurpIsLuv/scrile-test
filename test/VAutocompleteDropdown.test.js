import { mount, createLocalVue } from '@vue/test-utils'
import VAutocompleteDropdown from '@/components/common/VAutocompleteDropdown'

const localVue = createLocalVue()

describe('VAutocompleteDropdown', () => {
  it('Render correctly', () => {
    const wrapper = mount(VAutocompleteDropdown, { localVue })

    expect(wrapper.is(VAutocompleteDropdown)).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()
  })
})