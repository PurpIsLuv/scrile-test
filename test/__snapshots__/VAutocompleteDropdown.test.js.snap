// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VAutocompleteDropdown Render correctly 1`] = `
"<div class=\\"v-autocomplete-dropdown\\">
  <ol></ol>
</div>"
`;
